src/downloader.rs

```diff
@@ -325,13 +325,29 @@ impl Future for DownloadHandle {
     }
 }
 
+/// All numerical config options for the downloader.
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+pub struct Config {
+    /// Concurrency limits for the downloader.
+    pub concurrency: ConcurrencyLimits,
+    /// Retry configuration for the downloader.
+    pub retry: RetryConfig,
+}
+
 /// Handle for the download services.
-#[derive(Clone, Debug)]
+#[derive(Debug, Clone)]
 pub struct Downloader {
+    inner: Arc<Inner>,
+}
+
+#[derive(Debug)]
+struct Inner {
     /// Next id to use for a download intent.
-    next_id: Arc<AtomicU64>,
+    next_id: AtomicU64,
     /// Channel to communicate with the service.
     msg_tx: mpsc::Sender<Message>,
+    /// Configuration for the downloader.
+    config: Arc<Config>,
 }
```

Comment:
`Inner` is now wrapping `mpsc::Sender`, which is clonable, and `Arc<Config>`, which is also clonable.
But then it ends up inside an `Arc` in `Downloader`.
This seems redundant?
Why not
```rs
pub struct Downloader {
  next_id: Arc<AtomicU64>,
  msg_tx: mpsc::Sender<Message>,
  config: Arc<Config>,
}
```
?

Reply:
I still have the same number of arcs as before, since I am losing the arc around AtomicU64.

And having an inner means that a downloader handle is just a pointer, so it is not just cloneable but more cheaply cloneable.

The reason the config is in an Arc is so I can share it between the handle and deep down in the actual service, even though I don't do that yet.